### PR TITLE
Feature/azure dns secret in helm chart

### PR DESCRIPTION
**Pull request by ricogu**
deploy azureDNS secert as part of cert-manager deployment

### CODE REVIEW

**Review by SebastianBezold — APPROVED** [review]

File: environments/core/applicationsets/certmanager-applicationset.yaml [hunk 1]
```diff
@@ -27,13 +27,16 @@ spec:
...
       source:
-        helm:
-          valueFiles:
-            - values-certmanager.yaml
+        plugin:
+          name: argocd-vault-plugin-helm
+          env:
+            - name: AVP_SECRET
+              value: vault-secret
+            - name: helm_args
```
Comment (carslen) [inline environments/core/applicationsets/certmanager-applicationset.yaml:35]: Why referencing `values.yaml` in this way? The helm should use it by default, so there's no need to reference it. Also the var `helm_args` [won't be recognized](https://github.com/catenax-ng/k8s-cluster-stack/blob/main/apps/argocd/base/avp-argocd-cm.yaml#L11-L17) by AVP Plugin _argocd-vault-plugin-helm_.

File: apps/certmanager/templates/azureDNS-clientsecret.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    avp.kubernetes.io/path: "devsecops/data/azure/demo.catena-x.net/azuredns-config"
+  name: azuredns-config
+  namespace: cert-manager
+type: Opaque
+stringData:
+  VAULT_ADDR: https://vault.vault.demo.catena-x.net/
```
Comment (carslen) [inline apps/certmanager/templates/azureDNS-clientsecret.yaml:10]: This secret manifest looks weird. `VAULT_ADDR`, `AVP_TYPE` and `AVP_AUTH_TYPE` are not required as they belong to AVP configuration. _azuredns-config_ should only contain `client-secret` as key.

File: environments/core/applicationsets/certmanager-applicationset.yaml (hunk 1 above)
Comment (ricogu) [reply environments/core/applicationsets/certmanager-applicationset.yaml:35]: we want to provide a wau such as we can provide multiple values file, like `values-{{something}}.yaml`

File: apps/certmanager/templates/azureDNS-clientsecret.yaml (hunk 2 above)
Comment (ricogu) [reply apps/certmanager/templates/azureDNS-clientsecret.yaml:10]: will address this in https://github.com/catenax-ng/k8s-cluster-stack/pull/50

To be honest I am not sure the use of those parameter so just kept it

File: environments/core/applicationsets/certmanager-applicationset.yaml (hunk 1 above)
Comment (ricogu) [reply environments/core/applicationsets/certmanager-applicationset.yaml:35]: discussion will resume in pr: https://github.com/catenax-ng/k8s-cluster-stack/pull/50

File: apps/certmanager/templates/azureDNS-clientsecret.yaml (hunk 2 above)
Comment (ricogu) [reply apps/certmanager/templates/azureDNS-clientsecret.yaml:10]: resolved in the new pr

**Pull request merged**